luigi/contrib/slack_notifications.py

```diff
@@ -0,0 +1,148 @@
+
+'''
+Supports sending slack notifications for different task events.
+
+Usage:
+
+Create an incomming webhook on slack: https://api.slack.com/incoming-webhooks
+
+Use a configuration like the following on luigi.cfg
+.. code-block:: ini
+    [slack]
+    webhook: https://hooks.slack.com/services/...
+    events: ["START", "SUCCESS", "FAILURE"]
+    template: Luigi task {task_id} event {event}: {args} {kwargs}
+
+
+In the python module, setup the slack integration with:
+.. code-block:: python
+    from luigi.contrib.slack_notifications import configure_slack
+    configure_slack()
+'''
+
+
+import logging
+import luigi
+import os
+import json
+
+logger = logging.getLogger(__name__)
+
+try:
+    import requests
+except ImportError:
+    logger.warning("Module %s requires the python package 'requests'." % __name__)
+
+
+class slack(luigi.Config):
+    webhook = luigi.parameter.Parameter(
+        default='',
+        config_path=dict(section='slack', name='slack-webhook'),
+        description='Webhook URL to send the notifications')
+    channel = luigi.parameter.Parameter(
+        default='',
+        config_path=dict(section='slack', name='slack-channel'),
+        description='Channel to send slack notifications to')
+    username = luigi.parameter.Parameter(
+        default='',
+        config_path=dict(section='slack', name='slack-username'),
+        description='Slack username originating the message')
+    icon_emoji = luigi.parameter.Parameter(
+        default='',
+        config_path=dict(section='slack', name='slack-icon-emoji'),
+        description='Slack bot icon emoji')
+    events = luigi.parameter.ListParameter(
+        default=['START', 'FAILURE', 'SUCCESS'],
+        config_path=dict(section='slack', name='slack-events'),
+        description='List of events to be handled as slack notifications')
+    template = luigi.parameter.Parameter(
+        default='Luigi task {task_id} event {event}: {args} {kwargs}',
+        config_path=dict(section='slack', name='slack-template'),
+        description='Template message for event handling slack notifications')
+    event_templates = luigi.parameter.DictParameter(
+        default={},
+        config_path=dict(section='slack', name='slack-event-templates'),
```

Comment:
Yep. You should remove it. `[email]` is like that only to keep backwards compatibility. :)